src/model/Model.js

```diff
@@ -270,13 +270,36 @@ export default class Model {
 		return new KeyModel( escapeKey( this.key ) );
 	}
 
-	getKeypathModel () {
-		return this.keypathModel || ( this.keypathModel = new KeypathModel( this ) );
+	getKeypathModel ( ractive ) {
+		let keypath = this.getKeypath(), model = this.keypathModel || ( this.keypathModel = new KeypathModel( this ) );
+
+		if ( ractive && ractive.component ) {
+			let mapped = this.getKeypath( ractive );
+			if ( mapped !== keypath ) {
```

Comment:
Here, the rootpath (top-level keypath) is the same, but the keypath (mapped) is not. So a child `KeypathModel` is created and attached to the main `KeypathModel`. That lets children update appropriately when the main instance is shuffled, etc. Is this the correct way to handle the "keypath is the same but not really because it's also mapped" issue?